sympy/logic/boolalg.py

```diff
@@ -768,10 +769,13 @@ def _eval_simplify(self, **kwargs):
                         except ValueError:
                             pass
                     if x in reps:
-                        eqs.append(e.func(e.rhs, reps[x]))
-                    else:
+                        eqs.append(e.subs(x, reps[x]))
+                    elif e.lhs == x and x not in e.rhs.free_symbols:
                         reps[x] = e.rhs
                         eqs.append(e)
+                    else:
+                        # x is not yet identified, but may be later
+                        nonlineqs.append(e)
```

Comment:
I would suggest a slight reordering as:
```python
                        except ValueError:
                            # x is not yet identified, but may be later
                            nonlineqs.append(e)
                    elif x in reps:
                        eqs.append(e.subs(x, reps[x]))
                    else:
                        reps[x] = e.rhs
                        eqs.append(e)
```

Reply:
Makes sense!

Reply:
Ohh, but that would require 
```
reps[x] = e.rhs
eqs.append(e)
```
before `except ValueError` as well, right? Can be done and maybe clearer, but do you think it should be like that?

Reply:
No. I think what you have is right. The blocks cover, in order "either false", "one true", "other true", "both true".